### DISCUSSION TITLE
How can I build custom filters by components?

### DISCUSSION THREAD

**Discussion by z0091** [strapi/strapi#5894] [Legacy Topics]
Hi! 
I need your help. I have the following model. For simplicity, i removed all the extra fields.

```JSON
{
      "kind": "collectionType",
      "connection": "default",
      "collectionName": "lectures",
      "info": {
        "name": "lecture"
      },
      "options": {
        "increments": true,
        "timestamps": true
      },
      "attributes": {
        "access_type": {
          "type": "component",
          "repeatable": true,
          "component": "lecture.access-type",
          "required": true,
          "max": 2
        },
      },
    }
```

Strapi does not support filtering by component. I need to filter by `access_type.id`. I understand that I need to write my own queries to bookshelfjs. Do you have sample queries for filtering by component?
I worry that I will have to repeat the logic `node_modules/strapi-connector-bookshelf/lib/generate-component-relations.js`

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by lauriejim** [comment]
Hello! Connections made by Strapi doesn't allow filtering on this type of relation (polymorphic)
I suggest you use `await strapi.connections.default.raw('...');` to create a custom SQL query that will give you the data you want.
That is the easiest way I think.
